### PR TITLE
[TIR][TRANSFORM] Return value support in tir.tvm_call_packed

### PR DESCRIPTION
**Pull request by tqchen**
This PR fixes the return value support in tir.tvm_call_packed

- Clarified the semantics of the intrinsics
- Fix a problem when lowering call packed with nested scopes(let bindings)
- Added regression tests to cover the changes

### CODE REVIEW

**Comment by tqchen** [comment]
cc @csullivan @areusch @giuseros

File: tests/python/unittest/test_tir_transform_lower_tvm_builtin.py [hunk 1]
```diff
@@ -133,11 +133,45 @@ def check_packed_func(target="llvm"):
     tvm.ir.assert_structural_equal(alloca_shape, expected_stmt, map_free_vars=True)
 
 
-def test_packed_func():
+def test_lower_packed_func():
     check_packed_func("llvm")
```
Comment (giuseros) [inline tests/python/unittest/test_tir_transform_lower_tvm_builtin.py:137]: Can you add a check also for the C backend? I tried this in AOT, but the C backend appears to no handling return values correctly.

Comment (tqchen) [reply tests/python/unittest/test_tir_transform_lower_tvm_builtin.py:137]: i believe previously it was only works for non-C backend, but recent PR might added support

Comment (giuseros) [reply tests/python/unittest/test_tir_transform_lower_tvm_builtin.py:137]: Is the PR that adds non-C support already merged? If so, I think it should be ok to add a test. If not, we might add a TODO and add the test when the PR gets merged

Comment (tqchen) [reply tests/python/unittest/test_tir_transform_lower_tvm_builtin.py:137]: I agree a TODO is a good idea

**Review by csullivan — APPROVED** [review]
LGTM! Verified this with a packed function returning a handle. Thanks @tqchen

**Review by ZihengJiang — APPROVED** [review]

**Pull request merged**